test/CMakeLists.txt

```diff
@@ -1,93 +1,113 @@
 message(STATUS "Building Unit Tests ${UNITTEST}")
 
+# look for Python3
+find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+if (${PYTHON3_FOUND})
+    # use command 'pip show tensorflow' to check whether tensorflow is installed in python.
+    message(STATUS "Will use ${Python3_EXECUTABLE} to run scripts.")
+    message(STATUS "checking tensorflow")
+    execute_process(
```

Comment:
In case the user does not have TensorFlow installed, they will encounter a `ModuleNotFoundError: No module named 'tensorflow'` error anyways. Does this check really add much additional value?

Reply:
User should install and check tensorflow by themselves. And if tensorflow installed rightly, `execute_process` will set `EXIT_CODE=0`, if not, it will set `EXIT_CODE !=0`. And also we won't get `ModuleNotFoundError: No module named 'tensorflow'` by shell command `pip show tensorflow`.

Reply:
I think we just need to check whether tensorflow has been installed instead of checking whether it has been installed rightly. We also won't install tensorflow by cmake. Different platform has different installation method. Just let people install tensorflow by themselves and we only need to remind users to install tensorflow by message.

Reply:
Because users my do not necessarily read README.md carefully, they may directly read INSATLL.md (such as me), thus ignoring the installation and inspection of tensorflow. At the same time, there will be many non Python users who may not understand Python's error message about TensorFlow module `ModuleNotFoundError.` They may be pure C++programmers. `find_package` can remind the user to install Python, and the TensorFlow check can remind the user to install TensorFlow, otherwise the compilation cannot proceed.So I think it should be in CmakeLists.txt to provide a mandatory measure to remind the user to install TensorFlow. If the user does not install TensorFlow, Cmake cannot continue to work. It is not reminded by Python error messages in the 'make' stage.

Reply:
Ok, thanks a lot. I did not consider, that a `ModuleNotFoundError` error might not be enough. :+1:

I'm always careful about introducing additional complexity (especially in the auxiliary stuff, like CMake things), because of the additional maintenance, but you convinced me on that one. :slightly_smiling_face:

Reply:
I just tested the user experience when not having TensorFlow installed but trying to use frugally-deep:

```bash
git clone https://github.com/Dobiasd/frugally-deep
python3 ./frugally-deep/keras_export/convert_model.py
```

```
ModuleNotFoundError: No module named 'tensorflow.keras'

```

Then pasting `ModuleNotFoundError: No module named 'tensorflow.keras'` into Google immediately shows the correct, results, i.e., `pip install tensorflow`, etc.

This test changed my mind. I now think, having the proposed check in our `test/CMakeLists.txt` does not add enough value to be worth the increased complexity.

Reply:
Oh I agree with you,so shell we just show a message to remind user to check Tensorflow instead of we use `pip show tensorflow' in CmakeList.txt. I just remind that not all users run `pip` through `pip`, they might use `pip3`, `pip38'` or others, this will make mistakes. I think may be use a WARNNING message ?

Reply:
Yeah, just printing a message, reminding the user to have TensorFlow installed, would be ok. :+1:

I feel it should not be a warning though, because usually, a warning means:
"Hey, something is wrong, but it's not fatal, and luckily we can continue nonetheless."

In our case, however, we don't yet even know if something is wrong. Displaying a warning even if all is fine sounds misleading to me.

Reply:
I have just modified it and gave a commit.